Sources/ArgumentParser/Parsing/ArgumentSet.swift

```diff
@@ -315,6 +315,8 @@ extension ArgumentSet {
     
     try setInitialValues(into: &result)
     
+    let initialCounts = getInitialArrayCounts(in: result)
```

Comment:
I'm worried about this post-processing to remove the default values. Could you take care of emptying the array in the closure generated by `ArgumentDefinition.Update.appendToArray` instead?

Reply:
I moved the default clearing code to a custom `update` method on `ParsedValues` that is unique to arrays. Now it removes the defaults up front, when the first array value from the command line is parsed.